scripts/validate_response.py

```diff
@@ -0,0 +1,102 @@
+"""Response validation using pre-defined question/answer pair."""
+
+import argparse
+import json
+import sys
+
+import requests
+from llama_index.embeddings.huggingface import HuggingFaceEmbedding
+from scipy.spatial.distance import cosine, euclidean
+
+
+# TODO: Add more question/answer pair
+def args_parser(args):
+    """Arguments parser."""
+    parser = argparse.ArgumentParser(description="Response validation module.")
+    parser.add_argument(
+        "-s",
+        "--scenario",
+        choices=["with_rag", "without_rag"],
+        default="with_rag",
+        help="Scenario for which responses will be evaluated.",
+    )
+    parser.add_argument(
+        "-m",
+        "--model",
+        choices=["gpt", "granite"],
+        default="gpt",
+        help="Model for which responses will be evaluated.",
+    )
+    parser.add_argument(
+        "-t",
+        "--threshold",
+        type=lambda val: float(val),
+        default=0.25,
+        help="Threshold value to be used for similarity score.",
+    )
+    return parser.parse_args(args)
+
+
+def main():
+    """Validate llm response."""
+    args = args_parser(sys.argv[1:])
+    print(f"Arguments passed: {args}")
+
+    with open("./scripts/question_answer_pair.json") as qna_f:
```

Comment:
will it work if the script will be started as `python validate_response.py`? (ie from current dir)

Reply:
it might work from both directories. basically the script knows that the JSON is in the same directory as script itself. so it can use this info: `os.path.dirname(__file__)`